superbench/benchmarks/micro_benchmarks/cublaslt_function.py

```diff
@@ -75,20 +111,28 @@ def _preprocess(self):
 
         self.__bin_path = os.path.join(self._args.bin_dir, self._bin_name)
 
-        if self._args.in_type not in self._in_types:
-            logger.error(f'Invalid input type {self._args.in_type}.')
+        if not self.validate_mrange(self._args.batch):
+            logger.error(f'Invalid batch size {self._args.batch}.')
             return False
 
         self._commands = []
-        for shape in self._args.shapes:
-            shape_list = shape.replace(',', ' ').split()
-            if len(shape_list) != 3 or not all(x.isdigit() for x in shape_list):
-                logger.error(f'Invalid shape {shape}.')
+        for _in_type in self._args.in_types:
+            if _in_type not in self._in_types:
+                logger.error(f'Invalid input type {_in_type}.')
                 return False
-            self._commands.append(
-                f'{self.__bin_path} -m {shape_list[0]} -n {shape_list[1]} -k {shape_list[2]} '
-                f'-b {self._args.batch} -w {self._args.num_warmup} -i {self._args.num_steps} -t {self._args.in_type}'
-            )
+            for _b in self.mrange(*map(int, self._args.batch.split(':'))):
+                for shape in self._args.shapes:
+                    shape_list = shape.replace(',', ' ').split()
+                    if len(shape_list) != 3 or not all(self.validate_mrange(x) for x in shape_list):
+                        logger.error(f'Invalid shape {shape}.')
+                        return False
+                    for _m, _n, _k in itertools.product(
```

Comment:
what if m or n or k is 0?

Reply:
0 or 1 or other unsupported shapes will get "cuBLAS API failed" in runtime